### ISSUE TITLE
Process not exiting by default

### ISSUE THREAD

**Issue by philkunz** [nodegit/nodegit#866]
Running a js file that has has just one line that requires nodegit like so:
```javascript
var nodegit = require("nodegit");
```
results in the process not exiting... Is this a bug?
node in verson 4.2.4, nodegit in latest version from npm: 0.8.0

**Comment by johnhaley81** [comment]
Confirmed. This is a bug.

**Comment by 01AutoMonkey** [comment]
I'm also experiencing this

```node
var nodegit = require("nodegit");
console.log("exit?");
```

just hangs on `exit?`.

---

- nodegit@0.9.0
- node v5.0.0
- ubuntu 15.10

**Comment by johnhaley81** [comment]
@srajko can you look into this? I think this is related to #836

**Comment by lowellk** [comment]
Can anyone tell me the latest version that does not have this bug?

**Comment by johnhaley81** [comment]
This bug was introduced in 0.8.0 I believe.

**Comment by srajko** [comment]
Confirmed - this started happening with https://github.com/nodegit/nodegit/commit/80aabf234af0dd461916c7390fc350a18e3c1678

**Comment by srajko** [comment]
OK, I have a fix.

Thanks for the bug report @philkunz and @01AutoMonkey!

And thanks for pinging me @johnhaley81 :-)

**Comment by johnhaley81** [comment]
<3

**Comment by guoshencheng** [comment]
@srajko thanks, but, is the fix released ?

**Comment by tbranyen** [comment]
Check the README we have instructions for installing at various commits.  This won't install prebuilts though, so might not be useful for your needs.

**Comment by guoshencheng** [comment]
Sorry for I am a fresh man here，Do you mean [building from source](http://www.nodegit.org/guides/install/from-source/) ? So does It mean that I can't use `nodegit` with `npm install` ?

**Comment by guoshencheng** [comment]
temporarily, use 0.7.0 solved this problem

**Comment by philkunz** [comment]
just as a question: Why aren't patches released in a more constant manner to npm? This fix could be implemented in version 0.9.1 for example.

**Comment by johnhaley81** [comment]
Valid question.

Mostly because of the overhead in releasing that we have. Almost all related to updating `CHANGELOG.md` and https://www.nodegit.org. The building of the binaries is _mostly_ automated so that's not too bad anymore.

If those were eliminated then we could have a much faster release process which would help getting stuff out there faster.
